src/App/Fossa/EmbeddedBinary.hs

```diff
@@ -86,13 +87,18 @@ cleanupThemisBins (ThemisBins a b) = traverse_ cleanupExtractedBinaries [unTag a
 withThemisAndIndex :: Has (Lift IO) sig m => (ThemisBins -> m c) -> m c
 withThemisAndIndex = bracket extractThemisFiles cleanupThemisBins
 
+-- When running Themis we always need both the themis-cli and the decompressed index.gob
 extractThemisFiles :: Has (Lift IO) sig m => m ThemisBins
 extractThemisFiles = do
   themisActual <- applyTag @ThemisBinary <$> extractEmbeddedBinary Themis
-  -- TODO: The gob is still xzipped, unzip it here
-  themisIndex <- applyTag @ThemisIndex <$> extractEmbeddedBinary ThemisIndex
-
-  pure $ ThemisBins themisActual themisIndex
+  compressedThemisIndex <- extractEmbeddedBinary ThemisIndex
+  let decompressedThemisIndex =
+        BinaryPaths
+          { binaryPathContainer = binaryPathContainer compressedThemisIndex
+          , binaryFilePath = $(mkRelFile "index.gob")
+          }
+  sendIO $ extractLzma (toPath compressedThemisIndex) (toPath decompressedThemisIndex)
+  pure $ ThemisBins themisActual $ applyTag @ThemisIndex decompressedThemisIndex
```

Comment:
We have the entire binary file in memory, it seems a little weird to dump it out to a file and then unzip it on-disk.  I think it's best if we create some `extractXZippedBinary` function which does similar work to `extractEmbeddedBinary`, but runs in-memory `lzma` unzipping on the embedded bytestring and dumps that directly to the file.

Reply:
Looks like `Lzma.decompress` is what we're looking for.